pkg/crc/cluster/cluster.go

```diff
@@ -159,6 +159,11 @@ func RemovePullSecretFromCluster(ocConfig oc.Config, sshRunner *ssh.Runner) erro
 	if _, _, err := sshRunner.RunPrivileged("Removing user's pull secret", "rm", "-fr", vmPullSecretPath); err != nil {
 		return err
 	}
+	// Change the ownership of authorized_keys file
+	// https://bugzilla.redhat.com/show_bug.cgi?id=1956739
+	if _, _, err := sshRunner.RunPrivileged("Change the ownership of authorized_keys for core user", "chown", "core.core", "/home/core/.ssh/authorized_keys"); err != nil {
+		return err
+	}
```

Comment:
This does not belong in a method called `RemovedPullSecretFromCluster`

Reply:
Seems like we could change `updateSSHKeyPair` to use `sudo` to cover this MCO unexpected behaviour, and not need any changes here? We'd have all the code dealing with `authorized_keys` in a single place, which should make things slightly easier if we rework it to use the MCO to do its changes.